### ISSUE TITLE
Windows support: Don't try to use open_unix_connection under Windows-only Python

### ISSUE THREAD

**Issue by pyhedgehog** [ronf/asyncssh#68]
I'm using both windows-only flavour of python and cygwin, so I want to keep SSH_AUTH_SOCK env variable alone, but it ends with following traceback:
```
...
  File "forwarder.py", line 33, in start_forwarder
    conn = yield from asyncssh.connect(hostname)
  File "C:\tools\python35\lib\site-packages\asyncssh\misc.py", line 152, in __iter__
    return (yield from self._coro)
  File "C:\tools\python35\lib\site-packages\asyncssh\connection.py", line 4333, in connect
    conn, _ = yield from create_connection(None, host, port, **kwargs)
  File "C:\tools\python35\lib\site-packages\asyncssh\connection.py", line 4085, in create_connection
    agent = yield from connect_agent(agent_path)
  File "C:\tools\python35\lib\site-packages\asyncssh\agent.py", line 525, in connect_agent
    yield from agent.connect()
  File "C:\tools\python35\lib\site-packages\asyncssh\agent.py", line 157, in connect
    yield from asyncio.open_unix_connection(self._agent_path,
AttributeError: module 'asyncio' has no attribute 'open_unix_connection'
```

**Comment by ronf** [comment]
Thank you for this report! I knew Windows didn't support asyncio.open_unix_connection(), but didn't consider that someone might set SSH_AUTH_SOCK in the environment on a Windows system.

Since AsyncSSH silently ignore errors in connecting to an invalid SSH_AUTH_SOCK value today, I think the simplest fix for now is to do the same when this AttributeError occurs:

```patch
--- a/asyncssh/agent.py
+++ b/asyncssh/agent.py
@@ -524,5 +524,5 @@ def connect_agent(agent_path=None, *, loop=None):
     try:
         yield from agent.connect()
         return agent
-    except (OSError, ChannelOpenError):
+    except (OSError, ChannelOpenError, AttributeError):
         return None
```

I saw your other request to support the pageant protocol, and may need to do something different if I'm able to add support for that, but for now this should prevent the traceback you are seeing. I'll check this change into the "develop" branch shortly, and it will be part of the next release.

**Comment by pyhedgehog** [comment]
Thank you a lot.

**Comment by ronf** [comment]
This is now covered by the changes in #69, released in AsyncSSH 1.8.0.
